Makefile

```diff
@@ -0,0 +1,28 @@
+.PHONY: default
+default: help
+
+.PHONY: help
+help:
+	@echo "make help     Show this help message"
+	@echo "make dev      Run the plugin in the development WordPress"
+	@echo "make lint     Run the code linter(s) and print any warnings"
+	@echo "make format   Automatically format code"
+	@echo "make sure     Make sure that the formatter, linter, tests, etc all pass"
+
+.PHONY: dev
+dev: vendor
+	docker compose up
+
+.PHONY: lint
+lint: vendor
+	composer lint
```

Comment:
Is there any kind of PHP version management needed when installing dependencies, or is it like npm in our JS projects where we can get by without requiring a specific Node/npm version?

Reply:
The PHP version your project requires is defined as any other dependency, in the "require" block of the `composer.json` file. This block is the equivalent to `dependencies` in a package.json file, in the same way `require-dev` is the same as `devDependencies`.

https://github.com/hypothesis/wp-hypothesis/blob/main/composer.json#L11-L13

Unfortunately, it is only used to verify other dependencies you install will work with the PHP version you require, but it does not install the PHP binaries themeselves.